spectrum-ergo-connector/src/main.rs

```diff
@@ -0,0 +1,567 @@
+            config.http_client_timeout_duration_secs as u64,
+        ))
+        .build()
+        .unwrap();
+    let explorer_url = Url::try_from(String::from("https://api.ergoplatform.com")).unwrap();
+    let explorer = Explorer {
+        client: client.clone(),
+        base_url: explorer_url,
+    };
+
+    let node = ErgoNodeHttpClient::new(client, node_url);
+
+    let mut data_inputs = vec![];
+    for box_id in config.committee_box_ids {
+        let ergo_box = explorer.get_box(box_id).await.unwrap();
+        data_inputs.push(ergo_box);
+    }
+
+    let withdrawal_repo = WithdrawalRepoRocksDB::new(&config.withdrawals_store_db_path);
+    let vault_box_repo = VaultBoxRepoRocksDB::new(&config.vault_boxes_store_db_path);
+
+    let mut vault_handler = VaultHandler::new(
+        vault_box_repo,
+        withdrawal_repo,
+        config.committee_guarding_script,
+        config.committee_public_keys,
+        TxIoVec::try_from(data_inputs).unwrap(),
+        config.chain_sync_starting_height,
+        MovedValueHistoryRocksDB::new(&config.moved_value_history_db_path),
+        ExportTxRetrySchedulerRocksDB::new(
+            &config.export_tx_retry_config.db_path,
+            config.export_tx_retry_config.retry_delay_duration.num_seconds(),
+            config.export_tx_retry_config.max_retries,
+        )
+        .await,
+    )
+    .unwrap();
+
+    let bootstrapper = Bootstrapper::bind(config.unix_socket_path).unwrap();
+    let path = bootstrapper.path().to_owned();
+    let (msg_in_send, msg_in_recv) = symmetric_channel::<VaultRequest<ExtraErgoData>>().unwrap();
+    let (msg_out_send, msg_out_recv) =
+        symmetric_channel::<VaultResponse<ExtraErgoData, ErgoNotarizationBounds>>().unwrap();
+
+    enum StreamValueFrom {
+        Chain(TxEvent<(Transaction, u32)>),
+        Driver(VaultRequest<ExtraErgoData>),
+        ResubmitExportTx,
+    }
+
+    type CombinedStream = std::pin::Pin<Box<dyn futures::stream::Stream<Item = StreamValueFrom> + Send>>;
+
+    // Convert the tokio_unix_ipc Receiver into a stream.
+    let consensus_driver_stream = stream! {
+        loop {
+            if let Ok(msg) = msg_in_recv.recv().await {
+                yield msg;
+            }
+        }
+    };
+
+    // Every minute we check whether the export TX needs a resubmission.
+    let resubmit_export_tx_stream = stream! {
+        loop {
+            tokio::time::sleep(std::time::Duration::from_secs(60)).await;
+            yield ();
+        }
+    };
+
+    let streams: Vec<CombinedStream> = vec![
+        ReceiverStream::new(receiver).map(StreamValueFrom::Chain).boxed(),
+        consensus_driver_stream.map(StreamValueFrom::Driver).boxed(),
+        resubmit_export_tx_stream
+            .map(|_| StreamValueFrom::ResubmitExportTx)
+            .boxed(),
+    ];
+    let mut combined_stream = futures::stream::select_all(streams);
+
+    let participant_secret_keys = config.committee_secret_keys.clone();
+
+    // Mock consensus driver
```

Comment:
To ease testing, the mock consensus driver is spawned as another async task, which communicates over Unix socket. Currently it waits to sync with the connector, submits an export TX then waits for on-chain confirmation from the connector.